src/main/java/com/redhat/red/build/koji/KojiClient.java

```diff
@@ -708,6 +709,58 @@ public KojiImportResult importBuild( KojiImport importInfo, Iterable<Supplier<Im
         }, "Failed to retrieve list of tags for build: %s", buildId );
     }
 
+    /**
+     * Get tags giving a list of build Ids. This uses multicall and is much faster than calling listTags(id) one by one.
+     * @return A Map where the build Id is the key and tags as value ( a list ).
+     */
+    public Map<Integer, List<KojiTagInfo>> listTags( List<Integer> buildIds, KojiSessionInfo session )
+                    throws KojiClientException
+    {
+        MultiCallRequest multiCallRequest = new MultiCallRequest();
+        List<KojiMultiCallObj> multiCallObjs = new ArrayList<>();
+        multiCallRequest.setMultiCallObjs( multiCallObjs );
+
+        buildIds.forEach(( buildId ) -> {
+            KojiMultiCallObj callObj = new KojiMultiCallObj();
+            callObj.setMethodName( "listTags" );
+            List<Object> params = new ArrayList<>();
+            params.add( buildId );
+            callObj.setParams( params );
+            multiCallObjs.add( callObj );
+        });
+
+        List<KojiMultiCallValueObj> multiCallValueObjs = doXmlRpcAndWarn( () -> {
+            MultiCallResponse response =
+                            xmlrpcClient.call( multiCallRequest,
+                                               MultiCallResponse.class, sessionUrlBuilder( session ),
+                                               STANDARD_REQUEST_MODIFIER );
+
+            return response == null ? null : response.getValueObjs();
+        }, "Failed to retrieve tags multicall info for: %s", buildIds );
+
+        Map<Integer, List<KojiTagInfo>> ret = new HashMap<>(  );
+
+        Parser kojiTagInfoParser = Registry.getInstance().getParser( KojiTagInfo.class );
```

Comment:
Something we should think about doing is hiding this inside the Registry class. We could use `KojiTagInfo kojiTagInfo = Registry.getInstance().parseAs(o, KojiTagInfo.class)` instead.

Reply:
Ok.